src/Data/NonEmpty.purs

```diff
@@ -107,6 +104,8 @@ instance traversableWithIndexNonEmpty
 instance foldable1NonEmpty :: Foldable f => Foldable1 (NonEmpty f) where
   fold1 = foldMap1 identity
   foldMap1 f (a :| fa) = foldl (\s a1 -> s <> f a1) (f a) fa
+  foldr1 f (a :| fa) = maybe a (f a) $ foldr (\a1 -> Just <<< maybe a1 (f a1)) Nothing fa
```

Comment:
Shouldn't this just be? Why do we need the `Maybe`? If `fa` is empty, it'll just return `a`.
```purescript
foldr1 f (a :| fa) = foldr f a fa
```

Reply:
That would treat `a` as if it were the rightmost element, not the leftmost.

Reply:
Ha! I'm dumb. Duh!